File: backend/Hotel.java
package com.ahmedyassin.TravelSmart.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "hotels")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Hotel {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String address;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String country;

    @Column(nullable = false)
    private Integer etoile;

    @Column(name = "price_per_night")
    private Double pricePerNight;

    @Column(length = 1000)
    private String description;

    @Column(name = "image_url")
    private String imageUrl;

    @ElementCollection
    @CollectionTable(name = "hotel_amenities", joinColumns = @JoinColumn(name = "hotel_id"))
    @Column(name = "amenity")
    private List<String> amenities = new ArrayList<>();

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "is_active")
    private Boolean isActive = true;

    private Double latitude;
    private Double longitude;
    private String phone;
    private String email;
    private String website;

    @Column(name = "check_in_time")
    private String checkInTime;

    @Column(name = "check_out_time")
    private String checkOutTime;

    @Column(name = "cancellation_policy", length = 500)
    private String cancellationPolicy;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (isActive == null) {
            isActive = true;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: app/src/main/java/com/example/voyageproject/model/Hotel.kt
package com.example.voyageproject.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Hotel(
    val id: String,
    val name: String,
    val address: String,
    val city: String,
    val country: String,
    val etoile: Int,
    val pricePerNight: Double? = null,
    val imageUrl: String? = null,
    val description: String? = null,
    val amenities: List<String>? = null,
    val rating: Double = etoile.toDouble(),
    val rooms: List<Room>? = null,
    val phone: String? = null,
    val email: String? = null,
    val website: String? = null,
    val checkInTime: String? = null,
    val checkOutTime: String? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val isActive: Boolean? = true
) : Parcelable
